src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+    "react-error-boundary": "4.0.3"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+#### Providing an alternative value with `Promise.catch` {/*providing-an-alternative-value-with-promise-catch*/}
+
+If you'd like to display provide an alternative value when a promise passed to `use` is rejected you can use an the promise's [`catch`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch) method. To use the promise's `catch` method, call `catch` on the promise object. `catch` takes a single argument: a function that takes an error message as an argument. Whatever is returned by the function passed to `catch` will be used as the resolved value of the promise.
+
+<Sandpack>
+
+```js App.js
+import { useState, Suspense } from "react";
+import { Message } from "./message.js";
+
+function fetchMessage() {
+  return new Promise((resolve, reject) => {
+    setTimeout(reject, 3000, new Error("no new message found."));
+  }).catch((error) => {
+    console.log(error.toString());
+    return "no new message found.";
+  });
+}
+
+export default function App() {
+  const [messagePromise, setMessagePromise] = useState(null);
+  const [show, setShow] = useState(false);
+  function download() {
+    setMessagePromise(fetchMessage());
+    setShow(true);
+  }
+
+  if (show) {
+    return (
+      <Suspense fallback={<p>⌛Downloading message...</p>}>
+        <Message messagePromise={messagePromise} />
+      </Suspense>
+    );
+  } else {
+    return <button onClick={download}>Download new message</button>;
+  }
+}
+```
+
+```js message.js
+"use client";
+
+import { use } from "react";
+
+export function Message({ messagePromise }) {
+  const content = use(messagePromise);
+  return <p>Here is the message: {content}</p>;
+}
+```
+
+```js index.js hidden
+// TODO: update to import from stable
+// react instead of canary once the `use`
+// Hook is in a stable release of React
+import React, { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
+import "./styles.css";
+
+// TODO: update this example to use
+// the Codesandbox Server Component
+// demo enviornemnt once it is created
+import App from "./App";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+---
+
+## Troubleshooting {/*troubleshooting*/}
+
+### I'm getting this error: "An unsupported type was passed to use" {/*im-getting-this-error-an-unsupported-type-was-passed-to-use*/}
+
+The `use` hook only supports [JavaScript promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) and React's [Context](/learn/passing-data-deeply-with-context) as parameters.
```

Comment:
Does this error also come up when using a promise, but not a /supported/ shape of value of that promise? Like when we tried to pass the fetch itself, vs the json value of it

Reply:
I'm not sure I understand what you are asking. `use` will throw an error if you pass it anything but a promise or a context. If you pass JSON to `use` it will throw an error. I think we were confusing passing a promise to `use` vs passing an object.
```js
use(fetch('/some/api'); // works
```

```js
const response = fetch('/some/api')
use(response); // error
```

```js
const response = fetch('/some/api')
use(response.json()); // works
```

Reply:
I guess what I'm asking about is more is more around the serialization of promises over server components, vs what `use` supports. Promises that resolve to objects that contained functions couldn't be serialized obv so they failed when passed to `use`, right? But they would have failed anyways.

Reply:
Gotcha. Added a note in the Streaming section  and a troubleshooting section indicating that the resolved value of the promise must be serializable.